### PR TITLE
Make the ProgressDialog more readable

### PR DESCRIPTION
**Pull request by gismo2004**
Update progress bar and dialog

### CODE REVIEW

**Comment by heinrichI** [comment]
Why dialog not closed after search?
image

**Comment by Edi61** [comment]
I can confirm, there seem to be an error in implementation. Please fix it.

**Comment by gismo2004** [comment]
fixed

**Comment by Edi61** [comment]
I cannot confirm, still same. Problem is not related to closing, issue seems related to "Stopped".

**Comment by gismo2004** [comment]
Ok, can't reproduce this, at least I don't know how. However, I need to have a closer look on how it is meant to work and will try to find a better way to close the dialog properly.

**Comment by Edi61** [comment]
Hi @gismo2004
I rechecked it, now its closing, I have forgotten to update pull request sorry.

Ps.:
Maybe not caused by your change, but I am wondering, that progress bar seems to be finished too early.
In my example progress was 111 from 162. In the screenshot from @heinrichI it's 4211 from 4541.

File: src/AntiDupl.NET.WPF/View/ProgressDialog.xaml.cs [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
+
+    protected override void OnClosing(CancelEventArgs e)
     {
-        public ProgressDialog()
-        {
-            InitializeComponent();
-        }
+        base.OnClosing(e);
+
+        //find out, if the close button was pressed or if the window was closed by win.close()
```
Comment (heinrichI) [inline src/AntiDupl.NET.WPF/View/ProgressDialog.xaml.cs:23]: Is bad practise write anything in window file.

**Comment by heinrichI** [comment]
@gismo2004 what this code do? Why is it needed?
```
        //find out, if the close button was pressed or if the window was closed by win.close()
        if (new StackTrace().GetFrames().FirstOrDefault(x => x.GetMethod() == typeof(Window).GetMethod("Close")) != null) return;

        //if close button was pressed, cancel execution
        var vm = (ProgressDialogViewModel)DataContext;
        vm.CancelCommand.Execute(null);
        e.Cancel = true;
```

**Comment by gismo2004** [comment]
> 

Remove the code, run the program, close the dialog by clicking the win32 close button (right corner X) --> then you know...

File: src/AntiDupl.NET.WPF/View/ProgressDialog.xaml.cs (hunk 1 above)
Comment (gismo2004) [reply src/AntiDupl.NET.WPF/View/ProgressDialog.xaml.cs:23]: It is not! It depends on what has to be done!

**Comment by heinrichI** [comment]
@gismo2004 and?
https://user-images.githubusercontent.com/7752557/225687593-6baf62a6-f49c-4cd9-90ed-6dc6164b9ec6.mp4

**Comment by heinrichI** [comment]
@gismo2004 I got you.

**Pull request merged**